### ISSUE TITLE
3D Tensor ends as 2D Tensor

### ISSUE THREAD

**Issue by m-birke** [agaldran/torchvision_paired_transforms#1]
Hey,

I was excited to find your repo. Sadly I have got the following problem:

do you have any idea why the size of the 2nd image tensor is only 2D?

=============transforms============================

transform_list = [tr.ToTensor(),
                      tr.Normalize((0.5, 0.5, 0.5),
                                           (0.5, 0.5, 0.5))]
transform_list = [tr.RandomCrop((height, width))] + transform_list
transform_list = [tr.Resize(new_size)] + transform_list
transform_list = [tr.RandomHorizontalFlip()] + transform_list
transform = tr.Compose(transform_list)

img_tr1, img_tr2 = transform(test_img, test_img)
print(img_tr1.size())
print(img_tr2.size())

======output=========================================
torch.Size([3, 512, 512])
torch.Size([512, 512])

**Comment by m-birke** [comment]
I found the problem, It's because you made it work for grayscale targets. So colored targets / images doesn't work anymore

greetings

**Comment by agaldran** [comment]
Hi suitre77,

Yes, I changed that because you are most likely going to want to use "grayscale" images for labels in segmentation problems. The thing is that if you are using standard pytorch losses, like cross-entropy, those are going to expect you to have your labels in that format, where pixels holding a zero are for the first class, and so on. If your data comes in the form of "colored" targets, you will need to map all the possible colors you have in your data onto correlated integers. 

The previous version of this repo worked with "color" targets, so I guess you can still get that if you access previous versions. Good luck!

Adrian

**Comment by m-birke** [comment]
Hey Adrian,

thanks for your answer. ANd yes, I think my need was indeed special. I just forked it and adapted some small things to fit it to my use case.

Great repo, kind regards
